src/python/WMComponent/DBS3Buffer/MySQL/NewSubscription.py

```diff
@@ -40,15 +40,17 @@ def _createPhEDExSubBinds(self, datasetID, subscriptionInfo, custodialFlag):
 
         binds = []
         for site in sites:
-            binds.append({'id': datasetID,
-                          'site': site,
-                          'custodial': custodialFlag,
-                          'auto_approve': 1 if site in subscriptionInfo['AutoApproveSites'] else 0,
-                          'move': isMove,
-                          'priority': subscriptionInfo['Priority'],
-                          'phedex_group': phedex_group,
-                          'delete_blocks': delete_blocks,
-                          'dataset_lifetime': subscriptionInfo['DatasetLifetime']})
+            bind = {'id': datasetID,
```

Comment:
I think it would be more protective if you explicitly cast data-types for each binding value, e.g.
```
bind = {'id': int(datasetID), 'site': str(site), ...}
```
Doing this way you'll ensure that proper data-type is passed to bind dictionary which you later pass to ORACLE DB

Reply:
I think this is a good suggestion indeed.

Reply:
I'm updating this code with a few more safe checks for None. But these suggestions might actually cause more problems than resolve them. For instance, casting None to Integer will crash, casting a byte string to string will make things even worse.

In addition to that, `dataset_id` is already protected by the database schema and this code is in use for many years. Different than `dataset_lifetime`, which is a new feature.

Reply:
The way how you implemented those additional checks do look sound to me. Thanks @amaltaro